opennmt/utils/hooks.py

```diff
@@ -182,3 +183,33 @@ def end(self, session):
     tf.logging.info("Evaluation predictions saved to %s", self._output_path)
     if self._post_evaluation_fn is not None:
       self._post_evaluation_fn(self._current_step, self._output_path)
+
+class LoadCheckpointHook(tf.train.SessionRunHook):
+  """"Hook that loads model variables from checkpoint before starting the training."""
+
+  def __init__(self, checkpoint_path):
+    self.checkpoint_path = checkpoint_path
+
+  def begin(self):
+    var_list = tf.train.list_variables(self.checkpoint_path)
+    self.values = {}
+    for name, shape in var_list:
+      if not name.startswith("optim"):
+        if not name.startswith("global_step") and not name.startswith("words_per_sec"):
+          self.values[name] = np.zeros(shape)
+
+    reader = tf.train.load_checkpoint(self.checkpoint_path)
+    for name in self.values:
+      self.values[name] = reader.get_tensor(name)
+
+    tf_vars = []
+    current_scope = tf.contrib.framework.get_name_scope()
```

Comment:
Shouldn't this line get the current variable scope instead of the name scope?

```suggestion
    current_scope = tf.get_variable_scope()
```